app/models/sources/linode.rb

```diff
@@ -0,0 +1,161 @@
+module Sources
+  class Linode < Sources::Fog
+    class << self
+      def required_fields
+        super.merge({
+          api_key: :password,
+          root_password: :password,
+          image_name: :string,
+          flavor_name: :string,
+          kernel_name: :string,
+          data_center_name: :string
+        })
+      end
+    end
+
+    ID_TYPES = ['image', 'flavor', 'kernel', 'data_center']
+
+    # Connect to Linode to make sure our config is valid
+    def validate_config!
+      valid = false
+      begin
+        names_to_ids # Force re-evaluation of ids
+        valid = (!image_id.nil? && !flavor_id.nil? && !kernel_id.nil? && !data_center_id.nil?)
+      rescue Exception => e
+        add_error(e.message)
+      end
+      valid
+    end
+
+    private
+
+    # server_is_proxy_type?(server)
+    # given a server, determine if it is running a proxy
+    def server_is_proxy_type?(server)
+      # Fog implenentation does not give access to flavor/image/etc data
+      # So doing the most basic of checks here
+      return server.name =~ /proxy/
```

Comment:
You should add image/flavor/kernel/data_center info to the name then because otherwise you can't use the same linode account across two instances of Zartan or two Linode Source objects using different data_center_ids within the same Zartan instance.